### PR TITLE
Create dionaudio_loco-v2.c

### PR DESCRIPTION
**Pull request by miquel83blauw**

### CODE REVIEW

File: sound/soc/bcm/dionaudio_loco-v2.c [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+#include <sound/core.h>
+#include <sound/pcm.h>
+#include <sound/pcm_params.h>
+#include <sound/soc.h>
+#include <sound/jack.h>
+
+static bool digital_gain_0db_limit = true;
+
+static int snd_rpi_dionaudio_loco-v2_init(struct snd_soc_pcm_runtime *rtd)
```
Comment (pelwell) [inline sound/soc/bcm/dionaudio_loco-v2.c]: `-` is not legal character in identifiers - have you even tried to compile this?

**Comment by pelwell** [comment]
As well as the compilation issues, you will need a rule in the Makefile to build it, additions to Kconfig and the defconfigs to cause it to be built, and an overlay to load and configure it.

File: sound/soc/bcm/dionaudio_loco-v2.c [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+{
+	if (digital_gain_0db_limit) {
+		int ret;
+		struct snd_soc_card *card = rtd->card;
+		struct snd_soc_codec *codec = rtd->codec;
+
+		ret = snd_soc_limit_volume(codec, "Digital Playback Volume", 207);
```
Comment (pelwell) [inline sound/soc/bcm/dionaudio_loco-v2.c]: snd_soc_limit_volume takes a `struct snd_soc_card *` (i.e. card), so I don't know what line 35 is for.

**Comment by miquel83blauw** [comment]
Have updated the machine driver and overly. 

https://github.com/miquel83blauw/dionaudio-loco-v2

**Comment by pelwell** [comment]
Can you apply this patch (https://gist.github.com/pelwell/c09a905e0b645c11414b83a3d92510c9), see if it looks OK to you, then push it?

**Comment by miquel83blauw** [comment]
The patch looks ok. Sorry for asking, but how do I push it?

**Comment by pelwell** [comment]
Use `git am patchname` to apply the patch, then push it as normal. If that doesn't work for you, let me know and I'll push it directly to our 4.9 branch.

**Comment by miquel83blauw** [comment]
Sorry, I had tried multiple things but somehow it is not working for me...

**Comment by pelwell** [comment]
Don't worry - I'll push it. Can I mark the squashed patch as:
```
Signed-off-by: Miquel Blauw <miquelblauw@hotmail.com>
```
?

**Comment by miquel83blauw** [comment]
Thanks, Signing off by Miquel Blauw is ok.

Op 25 feb. 2017 om 17:41 heeft Phil Elwell <notifications@github.com<mailto:notifications@github.com>> het volgende geschreven:


Don't worry - I'll push it. Can I mark the squashed patch as:

Signed-off-by: Miquel Blauw <miquelblauw@hotmail.com<mailto:miquelblauw@hotmail.com>>


?

**Comment by pelwell** [comment]
Or would you prefer:
```
Signed-off-by: Miquel Blauw <info@dionaudio.nl>
```

**Comment by miquel83blauw** [comment]
Aa, I prefer: Miquel Blauw <info@dionaudio.nl<mailto:info@dionaudio.nl>>

Op 25 feb. 2017 om 17:51 heeft Phil Elwell <notifications@github.com<mailto:notifications@github.com>> het volgende geschreven:


Or would you prefer:

Signed-off-by: Miquel Blauw <info@dionaudio.nl<mailto:info@dionaudio.nl>>

**Comment by pelwell** [comment]
Merged manually.

**Pull request closed without merging**